Agent request cancel test

engelmi opened the pull request:
Fixes: https://github.com/eclipse-bluechi/bluechi/issues/796

Added an integration test to verify that shutting down the controller cancels all pending agent requests. It also moves the `Command` module to the new `bluechi_machine_lib` directory so that python scripts in the node can use it.

mkemel reviewed (approved):
LGTM

mkemel commented on tests/bluechi_machine_lib/command.py:
Are you sure about it? We are using this inside `bluechi_test`

engelmi replied:
Yes and no, the `Command` class makes sense in both cases - on the test runner and inside the machine/container. Actually, we have now two different `BluechiCtl` implementations... and the `SystemCtl` class would simplify a lot in the machine as well. 
So I am thinking the import flow would roughly be like `bluechi_machine_lib.*` -> `bluechi_test` with:
- `bluechi_machine_lib` being used on the local executor and machine
- `bluechi_test` only used on the local executor

But that should be implemented in another PR, probably. 
I re-evaluated the specific test case and could remove the python script.

mkemel commented on tests/tests/tier0/bluechictl-monitor-unit-created-removed/python/monitor.py:
```diff
@@ -125,8 +108,7 @@ def test_monitor_specific_node_and_unit(self):
         # Running bluechictl status on inactive unit should raise UnitCreated and UnitRemoved signals
-        res, out, _ = self.run_command(f"bluechictl status {node_name_foo} {service_simple}")
-        assert res == 0
+        out = Command(f"bluechictl status {node_name_foo} {service_simple}").run()
```
Why not just use `bluechi_machine_lib.bluechictl.BluechiCtl`?

mkemel replied:
Though having this module in `bluechi_machine_lib` does make a lot of sense, I would actually change BluechiCtl to use it

engelmi replied:
Yes, we have some duplication already with bluechictl. We need to fine-tune this a bit. But thats the focus of another PR :grin:

engelmi committed:
added integration test for agent_request_cancel

Fixes: https://github.com/eclipse-bluechi/bluechi/issues/796

Added an integration test to verify that shutting down the
controller cancels all pending agent requests.

Signed-off-by: Michael Engel <mengel@redhat.com>

mkemel reviewed (approved):
Nice solution!
LGTM

mwperina reviewed (approved):
LGTM
